src/core/build/types.ts

```diff
@@ -280,23 +280,25 @@ declare module "nitropack/types" {
     });
 
     for (const alias in tsConfig.compilerOptions!.paths) {
-      const paths = tsConfig.compilerOptions!.paths[alias];
-      tsConfig.compilerOptions!.paths[alias] = await Promise.all(
-        paths.map(async (path: string) => {
-          if (!isAbsolute(path)) {
-            return path;
-          }
-          const stats = await fsp
-            .stat(path)
-            .catch(() => null /* file does not exist */);
-          return relativeWithDot(
-            tsconfigDir,
-            stats?.isFile()
-              ? path.replace(/(?<=\w)\.\w+$/g, "") /* remove extension */
-              : path
-          );
-        })
+      const paths = new Set<string>(
+        await Promise.all(
+          tsConfig.compilerOptions!.paths[alias].map(async (path: string) => {
+            if (!isAbsolute(path)) {
+              return path;
+            }
+            const stats = await fsp
+              .stat(path)
+              .catch(() => null /* file does not exist */);
+            return relativeWithDot(
+              tsconfigDir,
+              stats?.isFile()
+                ? path.replace(/(?<=\w)\.\w+$/g, "") /* remove extension */
+                : path
+            );
+          })
+        )
       );
+      tsConfig.compilerOptions!.paths[alias] = [...paths];
```

Comment:
Code style, we could deduplicate paths directly here (this isn't immediately visible to me since we spit deduplicate (new Set + to array) in two places